File: trunk/quests/natimbi/#Cragbeast_Queen.pl
#BEGIN FILE: natimbi\#Cragbeast_Queen.pl (280080)

my $event = 0;

sub EVENT_SPAWN {
  quest::settimer("CBQ_Depop", 10);
  quest::ze(15,"A muffled roar echoes in the distance.");
}

sub EVENT_TIMER {
  if (($zonehour >= 2) && ($event == 0)) {
    quest::stoptimer("CBQ_Depop");
    quest::depop();
  }
}

sub EVENT_COMBAT {
  if ($combat_state == 0) {
    $event = 0;
  }
  else {
    $event = 1;
  }
}

sub EVENT_DEATH_COMPLETE {
  my $variance = 720; #Minute variance for respawn (Currently: 12 hours)
  my $new_time = int(rand($variance)) + 6840; #4 days 18 hours + 12 hour random to give us 5 day respawn with a 12 hour variance
  quest::setglobal("CBQ",1,3,"M$new_time");
}

#END of FILE zone:natimbi ID:280080 -- #Cragbeast_Queen.pl

File: trunk/quests/natimbi/#Cragbeast_Queen.lua
-- #Cragbeast_Queen NPCID: 280080
function event_spawn(e)
	eq.zone_emote(15,"A muffled roar echoes in the distance.");
end

function event_combat(e)
	if (e.joined) then
		-- turn off the despawn event if we go in to combat.
		eq.toggle_spawn_event(70,false,false)
	else
		-- turn it back on when we come out of combat
		eq.toggle_spawn_event(70,true,false)
		-- if the time is after 2am turn off the spawn condition that the spawn_event would have done
		local zone_time = eq.get_zone_time();
		if (zone_time["zone_hour"] >= 2 and zone_time["zone_hour"] < 24) then
			eq.spawn_condition("natimbi",0,1,0);
		end
	end
end

function event_death_complete(e)
	-- turn the depop spawn_event back on
	eq.toggle_spawn_event(70,true,false)
end
